### ISSUE TITLE
Did something change about how errors are added to the response in #3494

### ISSUE THREAD

**Issue by rmosolgo** [rmosolgo/graphql-ruby#3504]
cc @eapache , trying to replicate the change in behavior that you spotted. 

Here's a script that returns 1 error _before_ #3494 but 3 errors _after_ it. Gotta figure out why.

```ruby
require "bundler/inline"

gemfile do
  if ENV["BEFORE"]
    puts "using previous graphql-ruby ref"
    gem "graphql", github: "rmosolgo/graphql-ruby", ref: "b4a42cf8c012dee3d5866188b470b43b37c706be"
  elsif ENV["AFTER"]
    puts "using latest graphql-ruby ref"
    gem "graphql", github: "rmosolgo/graphql-ruby", ref: "master"
  else
    raise "Pass BEFORE=1 or AFTER=1"
  end
end

class BaseField < GraphQL::Schema::Field
  def authorized?(*)
    raise GraphQL::ExecutionError, "#{name} is not authorized"
  end
end

class Thing < GraphQL::Schema::Object
  field_class BaseField
  field :title, String, null: false
end

class Query < GraphQL::Schema::Object
  field :things, [Thing], null: false

  def things
    [{title: "a"}, {title: "b"}, {title: "c"}]
  end
end

class Schema < GraphQL::Schema
  query Query
end

query = GraphQL::Query.new(Schema, "{ things { title } }")
res = query.result
puts "GraphQL: #{GraphQL::VERSION}"
pp res["errors"].size
pp res.to_h

```

**Comment by toneymathews** [comment]
This behaviour still seem to exist in some cases. Unlike the test case written in https://github.com/rmosolgo/graphql-ruby/pull/3505, the `ExecutionError` gets replicated when raised on a field defined within an object when it's a connection. 

This [test case](https://gist.github.com/toneymathews/bffd9c66d43701be2002b4e4a5e3f0b6) was meant to replicate it within the gem, but there was only one error added to the response. Could there be another edge case where the query execution continues and updates the response even after marked as `graphql_dead`?

**Comment by rmosolgo** [comment]
It's certainly possible ... can you share any other details (eg, example query string) about the situation where you encounter this?

**Comment by toneymathews** [comment]
Yep, I've replicated it. It happens when there is additional field level authorization check. It's detailed in [this gist](https://gist.github.com/toneymathews/108ced30a3e7f06436717b65f49cb3a0)

**Comment by toneymathews** [comment]
I got that a bit wrong 🤦🏼 . If I change[ this existing test](https://github.com/rmosolgo/graphql-ruby/blob/master/spec/graphql/execution/interpreter_spec.rb#L627-L657) ( or the test in the issue description) to run for a connection rather than a non null list, the execution errors get repeated, rather than just appearing once in the response. ([Gist](https://gist.github.com/toneymathews/a16eb572d9dce19ba6812ff9342895b0))

**Comment by swalkinshaw** [comment]
I figured out the difference here and it's due to non-nullable node and edge fields.

```ruby
require "bundler/inline"

gemfile(true) do
  gem "graphql", "1.12.10"
end

p GraphQL::VERSION

module ConnectionErrorTest
  class BaseField < GraphQL::Schema::Field
    def authorized?(obj, args, ctx)
      raise GraphQL::ExecutionError, "#{name} is not authorized"
    end
  end

  class BaseConnection < GraphQL::Types::Relay::BaseConnection
    node_nullable(false)
    edge_nullable(false)
    edges_nullable(false)
  end

  class BaseEdge < GraphQL::Types::Relay::BaseEdge
    node_nullable(false)
  end

  class BaseObject < GraphQL::Schema::Object
    connection_type_class BaseConnection
    edge_type_class BaseEdge
  end


  class Thing < BaseObject
    field_class BaseField
    field :title, String, null: false
  end

  class Query < GraphQL::Schema::Object
    field :things, Thing.connection_type, null: false

    def things
      [{title: "a"}, {title: "b"}, {title: "c"}]
    end
  end

  class Schema < GraphQL::Schema
    query Query
  end
end

pp ConnectionErrorTest::Schema.execute("{ things { edges { node { title } } } }").to_h
```

This returns:

```ruby
{"data"=>nil,
 "errors"=>
  [{"message"=>"title is not authorized",
    "locations"=>[{"line"=>1, "column"=>27}],
    "path"=>["things", "edges", 0, "node", "title"]}]}
```

Note that `#authorized?` is called _3_ times, but only 1 error is in the output.

**Comment by rmosolgo** [comment]
Awesome, thanks for sorting that out! I've got a fix in the works at #3601. Can y'all give that a try to make sure it works for you? (Or, if not, I can merge and release, if you'd rather.)

**Comment by swalkinshaw** [comment]
Confirmed it fixes the issue 👍  thanks

**Comment by rmosolgo** [comment]
(🚢 'd a fix in 1.12.16, thanks again!)
